schrift.c

```diff
@@ -843,13 +843,52 @@ cmap_fmt6(SFT_Font *font, uint_fast32_t table, SFT_UChar charCode, SFT_Glyph *gl
 	return 0;
 }
 
+static int
+cmap_fmt12_13(SFT_Font *font, uint_fast32_t table, SFT_UChar charCode, SFT_Glyph *glyph, int which)
+{
+	uint32_t len, numEntries;
+	uint_fast32_t i;
+
+	*glyph = 0;
+
+	if (!is_safe_offset(font, table, 4))
```

Comment:
**Buffer Overrun:** This line only checks whether `table` contains at least 4 bytes.
But since we read 4 bytes starting at offset 4, we need to check for at least 8.
Since the table header must be at least 16 bytes long, we may as well check for 16.

Reply:
Your `cmap_fmt12_13()` routine scans through the entries naively.
If there aren't too many entries, this is fine.
But since TrueType guarantees that the entries are sorted by ascending `startCharCode`, this can be optimized with a binary search.